blocks/rapid-result/rapid-result.js

```diff
@@ -25,31 +31,35 @@ function formatDate(date) {
   return result;
 }
 
-async function decorateCards(block, placeholders, cardCount, previousNode) {
-  const queryObj = await fetchRapidResultMockData();
-  if (!queryObj) {
-    const mainWrapper = block.querySelector('.main-wrapper');
-    const mainContent = document.createElement('div');
-    mainContent.className = 'no-results-container';
-    mainWrapper.insertBefore(mainContent, previousNode);
-    handleNoResults(mainContent);
-    return;
-  }
-  const results = queryObj.map((el) => {
-    const elArr = el.PublishedOnDate.split(' ');
-    const publishDate = elArr[0];
-    const publishTime = elArr[1];
-    const formatPublishDate = publishDate.split('-').reverse();
-    el.PublishedOnDate = `${formatPublishDate} ${publishTime}`;
-    return el;
-  }).sort((a, b) => {
-    const dateA = new Date(a.PublishedOnDate);
-    const dateB = new Date(b.PublishedOnDate);
+function sortResult(sortData) {
+  const results = sortData.sort((a, b) => {
+    const dateA = new Date(a.pubDate);
+    const dateB = new Date(b.pubDate);
     return dateB - dateA;
   });
+  return results;
+}
+
+function generateViewMoreLink(name) {
+  return typeof name === 'string'
+    ? name
+      .toLowerCase()
+      .replace(/[^0-9a-z\s]/gi, '')
+      .replace(/\s/g, '-')
+    : '';
+}
+
+function getLink(stockName) {
+  return SITE_ROOT + RAPIDRESULT_DIRECTORY + generateViewMoreLink(stockName);
+}
 
-  const ul = createElement('ul', '');
+function buildCards(results, cards, cardCount, placeholders) {
   const loopNum = results.length > cardCount ? cardCount : results.length;
+  if (loopNum === 1) {
+    cards.classList.add('central-cards-1');
```

Comment:
@JiangLong2019  can this be handled with only child selector instead of two different css class
something like 
```
.parent-element > .child-element:only-child {
  /* Your CSS rules here */
}
```

Reply:
@anagarwa Is there any good property to handle the following case?
`When the parent ul has only one child li, add some stylings to the ul`

Reply:
@JiangLong2019 will this work for you

`
<style>
ul:has(:only-child) { color:green }
</style>
<html>
<ul>
abc
<li>xyz</li>
</ul>
</html>
`

Reply:
@anagarwa Unfortunately, I tried the same before, it didn't work. That's why I added an additional class.